django/db/models/fields/__init__.py

```diff
@@ -248,7 +246,7 @@ def _check_choices(self):
         if not is_iterable(self.choices) or isinstance(self.choices, str):
             return [
                 checks.Error(
-                    "'choices' must be an iterable (e.g., a list or tuple).",
+                    "'choices' must be a dictionary or an iterable (e.g., a list or tuple).",
```

Comment:
You're right - this is a hard one to word. A dictionary is an iterable, but we care about the fact that it's a `Mapping` and the fact it's iterable works against us.

I've re-worded it to say `'choices' must be a mapping (e.g. a dictionary) or an iterable (e.g., a list or tuple).`. I'm not thrilled about that - 99.9% of use cases will just be a plain dictionary, adding the `mapping` terminology into the mix makes it less clear IMO - even though yes, a dictionary is technically also an iterable.

Reply:
Do we want to prefer "dict" instead of "dictionary"?

You could change "iterable" to "sequence" which -- looking at the inheritance of ABC's in [`collections.abc`](https://docs.python.org/3/library/collections.abc.html) -- would exclude "mapping". It also would exclude "set" which is also considered "iterable" but not documented for use with choices. (We wouldn't want sets anyway, as it wouldn't maintain order.) Obviously, a change to "sequence" would then preclude "generator" which is also covered by "iterable". I don't know whether we currently support generators in `Field.choices`?